File: occasum-core/src/main/java/pw/stamina/occasum/properties/primitives/clamp/IntClamp.java
package pw.stamina.occasum.properties.primitives.clamp;

public interface IntClamp {

    int clamp(int value);

    static IntClamp min(int min) {
        return new MinIntClamp(min);
    }

    static IntClamp max(int max) {
        return new MaxIntClamp(max);
    }

    static IntClamp range(int min, int max) {
        return new RangeIntClamp(min, max);
    }

    static IntClamp none() {
        return NoClamp.INSTANCE;
    }
}

File: occasum-core/src/main/kotlin/pw/stamina/occasum/properties/primitives/clamp/IntClamp.kt
package pw.stamina.occasum.properties.primitives.clamp

interface IntClamp {

    fun clamp(value: Int): Int

    companion object {

        fun min(min: Int): IntClamp = MinIntClamp(min)

        fun max(max: Int): IntClamp = MaxIntClamp(max)

        fun range(min: Int, max: Int): IntClamp = RangeIntClamp(min, max)

        fun none(): IntClamp = NoClamp
    }
}
